### PR TITLE
Rerender and pin versions

### LINKED ISSUE #10: OSX build failing
https://travis-ci.org/conda-forge/matplotlib-feedstock/jobs/114864238

https://github.com/conda-forge/matplotlib-feedstock/pull/9#issuecomment-202030060

```
===== testing package: matplotlib-1.5.1-np110py27_0 =====
import: 'matplotlib'
import: 'matplotlib.pyplot'
Traceback (most recent call last):
  File "/Users/travis/miniconda3/conda-bld/test-tmp_dir/run_test.py", line 28, in <module>
    import matplotlib.pyplot
  File "/Users/travis/miniconda3/envs/_test/lib/python2.7/site-packages/matplotlib/pyplot.py", line 29, in <module>
    import matplotlib.colorbar
  File "/Users/travis/miniconda3/envs/_test/lib/python2.7/site-packages/matplotlib/colorbar.py", line 34, in <module>
    import matplotlib.collections as collections
  File "/Users/travis/miniconda3/envs/_test/lib/python2.7/site-packages/matplotlib/collections.py", line 27, in <module>
    import matplotlib.backend_bases as backend_bases
  File "/Users/travis/miniconda3/envs/_test/lib/python2.7/site-packages/matplotlib/backend_bases.py", line 62, in <module>
    import matplotlib.textpath as textpath
  File "/Users/travis/miniconda3/envs/_test/lib/python2.7/site-packages/matplotlib/textpath.py", line 15, in <module>
    import matplotlib.font_manager as font_manager
  File "/Users/travis/miniconda3/envs/_test/lib/python2.7/site-packages/matplotlib/font_manager.py", line 58, in <module>
    from matplotlib import ft2font
ImportError: dlopen(/Users/travis/miniconda3/envs/_test/lib/python2.7/site-packages/matplotlib/ft2font.so, 2): Library not loaded: libfreetype.6.dylib
  Referenced from: /Users/travis/miniconda3/envs/_test/lib/python2.7/site-packages/matplotlib/ft2font.so
  Reason: image not found
```

### LINKED ISSUE #14: Build/run dependencies
`libpng` is listed as a dependency on all platforms at build time, but `unix` only at run time.

### PR DESCRIPTION
**Pull request by ocefpaf**
- should help https://github.com/conda-forge/wradlib-feedstock/pull/9

Closes:
- [X] #14 (Build/run dependencies)
- [X] #10 (OS X failure)
- [ ] #11 (Windows build)

### CODE REVIEW

**Comment by ocefpaf** [comment]
I am skipping Windows64 and Python <3.5to avoid the

`src/ft2font.cpp(5) : fatal error C1083: Cannot open include file: 'string': No such file or directory`

error. I found [this](https://social.msdn.microsoft.com/Forums/vstudio/en-US/0f580a73-6f11-4aa6-abbe-6bf1438734c2/standard-files-stddefh-stringh-missing-in-visual-studio-2015-rc), but I was not successful. When I changed the `<string.h>` to just `<string>` new errors emerged.

TL;DR I think this is "good enough" and the Win64 issue should be addressed in another PR by someone that knows Windows more than I do.

Ping: @patricksnape and @gillins any ideas?

**Comment by janschulz** [comment]
AFAIK the conda recipe in matplotlib itself does not have this problem.

https://ci.appveyor.com/project/mdboom/matplotlib/build/1.0.1437/job/89u38439shestw4b
https://github.com/matplotlib/matplotlib/tree/master/ci/conda_recipe

**Comment by gillins** [comment]
Is this the same error: https://github.com/conda-forge/matplotlib-feedstock/issues/1 ?
Seems weird that it builds fine with Python 3.5.

**Comment by ocefpaf** [comment]
> AFAIK the conda recipe in matplotlib itself does not have this problem.

Thanks for pointing that out @janschulz. The recipe per se is virtually identical. I cannot see any obvious reason for the problem. We do have a different pinning/channel combination. Also, you are building latest source, right? (You did say in https://github.com/conda-forge/matplotlib-feedstock/issues/1#issuecomment-170583858 that you could replicate this bug on Python 3.4, is that still true?)

> Is this the same error: #1 ?

Yep. Same bug.

> Seems weird that it builds fine with Python 3.5.

Annoying isn't it :unamused: 

BTW I am merging this as-is and opening a new issue to resolve the Windows builds later. This PR already fixes two issues.

**Comment by ocefpaf** [comment]
For the renaming Windows issue see https://github.com/conda-forge/matplotlib-feedstock/issues/1.

**Comment by janschulz** [comment]
@ocefpaf No, that was before I rebuild the recipe there: https://github.com/matplotlib/matplotlib/pull/5629#issuecomment-174505476 and reading the comments there: https://github.com/matplotlib/matplotlib/pull/5629#issuecomment-174628531
-> It seems that this was a problem of wrong the wrong vsvargs batch file called?

from https://github.com/matplotlib/matplotlib/pull/5629#issuecomment-174742799
> Ok the 3.4 build is hopefully fixed by using %PYTHON% instead of python... No clue why this could happen...

**Comment by ocefpaf** [comment]
Thanks @janschulz!
